### PR TITLE
CSS Value Input that combines the TextField and Combobox updates all in one

### PR DESCRIPTION
**Pull request by kof**
- [x] use new combobox and textfield in props
- [ ] fix visual regressions in style panel

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/props-panel/props-panel.tsx
Comment (okonet) [inline apps/designer/app/designer/features/props-panel/props-panel.tsx:65]: I don’t like this API. They hide the real thing with no benefit. What is primitive and what it’s doing? What is anchor? Keep it as is by using Popper?

File: apps/designer/app/designer/features/props-panel/props-panel.tsx
Comment (okonet) [inline apps/designer/app/designer/features/props-panel/props-panel.tsx:79]: I’d move this to props getter to be safe you don’t override what’s inside.

File: apps/designer/app/designer/features/props-panel/props-panel.tsx
Comment (kof) [reply apps/designer/app/designer/features/props-panel/props-panel.tsx:65]: I could call it ComboboxPopper etc to make it more clear, I think it makes sense to reexport just to make sure we can do something in combobox later if needed

Comment (kof) [reply apps/designer/app/designer/features/props-panel/props-panel.tsx]: changed

File: apps/designer/app/designer/features/props-panel/props-panel.tsx
Comment (kof) [reply apps/designer/app/designer/features/props-panel/props-panel.tsx]: done

File: apps/designer/app/designer/features/props-panel/props-panel.tsx
Comment (okonet) [inline apps/designer/app/designer/features/props-panel/props-panel.tsx:65]: I still don't see any value in re-exporting at least not at this point but I don't care that much about it so it's okay.

File: packages/design-system/src/components/combobox.tsx
Comment (okonet) [inline packages/design-system/src/components/combobox.tsx:121]: I think it's a stupid rule to have. What's wrong with function keyword? How is the const better considering the ugly TS syntax?

File: packages/design-system/src/components/combobox.tsx
Comment (okonet) [inline packages/design-system/src/components/combobox.tsx:157]: `resetFilter` would read better IMO.

**Review by okonet — APPROVED** [review]
Feel free to merge as all my comments are stylistic.

File: packages/design-system/src/components/combobox.tsx
Comment (kof) [reply packages/design-system/src/components/combobox.tsx:121]: Another option is to enable Item extends unknown in the linter, so that it looks it least less like a hack. If you really want to discuss function keyword, we can do that separately

**Comment by kof** [comment]
@okonet I am going to need to fix the styling still, before I merge, it still looks broken

File: packages/design-system/src/components/combobox.tsx
Comment (okonet) [reply packages/design-system/src/components/combobox.tsx:121]: I actually don't want to discuss code style since it has 0 or even negative value. Spending time on fixing such issues just because we have this rule is a waste of everyone's time IMO. But it looks like you have a strong option about it so let's keep doing what we doing 🤷🏻‍♂️

Comment (kof) [reply packages/design-system/src/components/combobox.tsx:121]: I agree, it doesn't really matter, I feel strongly about it because I am trying to enforce a more consistent structure of code, meaning the order of definition - first define, then use. With named functions people can create very different structure and make it harder to read overall, you can't scan the code consistently.
When a few people work on the project - it doesn't matter. At scale this flexibility makes it harder to read.
In this particular case it wouldn't matter, because its a small file that will stay small etc. But someone will look it up and do the same somewhere else, where its not a good thing any more. So I am trying to set an example.

**Comment by kof** [comment]
@okonet going to merge this after last check from you, combobox in style is still visually wrong, but will get rid of it by porting the work from other PRs

**Pull request merged**
